src/backend/distributed/deparser/deparse_seclabel_stmts.c

```diff
@@ -0,0 +1,78 @@
+/*-------------------------------------------------------------------------
+ *
+ * deparse_seclabel_stmts.c
+ *	  All routines to deparse SECURITY LABEL statements.
+ *
+ * Copyright (c), Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "distributed/deparser.h"
+#include "nodes/parsenodes.h"
+#include "utils/builtins.h"
+
+static void AppendSecLabelStmt(StringInfo buf, SecLabelStmt *stmt);
+
+/*
+ * DeparseSecLabelStmt builds and returns a string representing of the
+ * SecLabelStmt for application on a remote server.
+ */
+char *
+DeparseSecLabelStmt(Node *node)
+{
+	SecLabelStmt *secLabelStmt = castNode(SecLabelStmt, node);
+	StringInfoData buf = { 0 };
+	initStringInfo(&buf);
+
+	AppendSecLabelStmt(&buf, secLabelStmt);
+
+	return buf.data;
+}
+
+
+/*
+ * AppendSecLabelStmt generates the string representation of the
+ * SecLabelStmt and appends it to the buffer.
+ */
+static void
+AppendSecLabelStmt(StringInfo buf, SecLabelStmt *stmt)
+{
+	appendStringInfoString(buf, "SECURITY LABEL ");
+
+	if (stmt->provider != NULL)
+	{
+		appendStringInfo(buf, "FOR %s ", stmt->provider);
```

Comment:
Does that need quotes?

Reply:
Yes, either `quote_literal_cstr` or `quote_identifier` should be used here (not sure which one).

Reply:
Not sure if I got your point, but the provider name is without quotes.
The syntax is, for example: `SECURITY LABEL for citus_tests_label_provider ON ROLE user1 IS 'citus_classified'; `
`citus_tests_label_provider` and `user1` don't have quotes, only the label.

Reply:
Sorry, I should've been more clear; do we need to call one of those quote functions to escape the characters that need escaping?

Reply:
Hmmm you are probably right about that, I will add the quotes and also tests with characters that need escaping.

Reply:
You are right, `quote_identifier` needs to be used.